app/jinja_filters.py

```diff
@@ -75,12 +75,18 @@ def format_unit(
     value: int | float | Decimal,
     length: UnitLengthType = "short",
 ) -> str:
+    # mass-metric-ton no longer supported for en_GB and related locales, but still present in business schema and allowed in validator,
+    # until removed from schema we substitute mass-ton for mass-metric-ton and vice versa, before/after unit formatting
+    measurement_unit = "mass-ton" if unit == "mass-metric-ton" else unit
```

Comment:
I think instead we need to replace `mass-metric-ton` with `mass-tonne` and then the second replacement after calling `units.format_unit` won't be required

Reply:
But only `mass-ton` is supported in validator atm, so no schema with `mass-tonne` is possible.

Reply:
Well, this will simplify it a lot in both.